pkg/scheduler/queue/tenant_queues.go

```diff
@@ -117,42 +117,88 @@ func newQueueBroker(maxUserQueueSize int, forgetDelay time.Duration) *queueBroke
 			tenantsByID:        map[TenantID]*queueTenant{},
 			tenantQuerierIDs:   map[TenantID]map[QuerierID]struct{}{},
 		},
-		maxUserQueueSize: maxUserQueueSize,
+		maxTenantQueueSize: maxTenantQueueSize,
 	}
 }
 
 func (qb *queueBroker) len() int {
 	return len(qb.tenantQueues)
 }
 
-// Returns existing or new queue for user.
-// MaxQueriers is used to compute which queriers should handle requests for this user.
-// If maxQueriers is <= 0, all queriers can handle this user's requests.
+func (qb *queueBroker) enqueueRequestBack(r requestToEnqueue) error {
+	queue, err := qb.getOrAddTenantQueue(r.tenantID, r.maxQueriers)
+	if err != nil {
+		return err
+	}
+
+	if queue.Len()+1 > qb.maxTenantQueueSize {
+		return ErrTooManyRequests
+	}
+
+	queue.PushBack(r.req)
+	return nil
+}
+
+func (qb *queueBroker) enqueueRequestFront(r requestToEnqueue) error {
```

Comment:
It looks like the queue size limit isn't being enforced here. Can you add a comment explaining that this is on purpose because this method should only be called to retry failures (if that's accurate)?